src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -93,18 +100,29 @@ private static Method getFireAuditStartedMethod() throws NoSuchMethodException {
         return fireAuditStarted;
     }
 
+    private static Method getAcceptFileStartedMethod() throws NoSuchMethodException {
+        final Class<Checker> checkerClass = Checker.class;
+        final Method acceptFileStarted =
+            checkerClass.getDeclaredMethod("acceptFileStarted", String.class);
+        acceptFileStarted.setAccessible(true);
+        return acceptFileStarted;
+    }
+
     @Test
     public void testDestroy() throws Exception {
         final Checker checker = new Checker();
         final DebugAuditAdapter auditAdapter = new DebugAuditAdapter();
         checker.addListener(auditAdapter);
         final DebugFilter filter = new DebugFilter();
         checker.addFilter(filter);
+        final TestBeforeExecutionFileFilter fileFilter = new TestBeforeExecutionFileFilter();
+        checker.addBeforeExecutionFileFilter(fileFilter);
 
         // should remove al listeners and filters
         checker.destroy();
 
         // Let's try fire some events
+        getAcceptFileStartedMethod().invoke(checker, "Some File Name");
```

Comment:
not sure how that bad examples of reflection are appeared here.
but all that could be invoked by  - `checker.process(.....)`